### PR TITLE
Bootstrap 3 theme for client interface

### PR DESCRIPTION
**Pull request by slm4996**
Updated 3-18-15!

Client side is moslty done, I am continuing to migrate the stylesheets over to bootstrap as I get time. Here are the latest screenshots.

Desktop:
osticket-bootstrap-desktop

Mobile:
osticket-bootstrap-mobile

### CODE REVIEW

**Comment by slm4996** [comment]
Commit  cc75b58 starts fresh with the client theme and bootstrap3 based on the latest develop codebase. This should be a clean import, This is fully useable as is and matches the current theme and style. I will continue as time allows to further integrate the theme into the bootstrap stylesheet and away from the in-built "osTicket" stylesheet for ease of maintenance moving forward.

**Comment by slm4996** [comment]
Commit 04bd748 : 
osticket-bootstrap-kb

**Comment by greezybacon** [comment]
I would like the see the bootstrap files fetch and packaged with the `setup/cli/package.php` script which we use to build the releases. You might use the `setup/cli/modules/i18n.php` file for some auto HTTP download idea code. This way we're not stashing upstream code we don't intend to modify in our repo

**Comment by slm4996** [comment]
I'll check into that @greezybacon. With this latest push I'd like some general feedback on the overall direction of what I've done before I push on.

File: include/client/header.inc.php [hunk 1]
```diff
@@ -21,19 +21,19 @@
     <meta name="description" content="customer support platform">
     <meta name="keywords" content="osTicket, Customer support system, support ticket system">
     <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
-	<link rel="stylesheet" href="<?php echo ROOT_PATH; ?>css/osticket.css" media="screen">
+    <link rel="stylesheet" href="<?php echo ROOT_PATH; ?>assets/bootstrap-dist/css/bootstrap.min.css" rel="stylesheet">
+    <link rel="stylesheet" href="<?php echo ROOT_PATH; ?>css/osticket.css" media="screen">
     <link rel="stylesheet" href="<?php echo ASSETS_PATH; ?>css/theme.css" media="screen">
+    <!--<link rel="stylesheet" href="<?php echo ASSETS_PATH; ?>css/bootstrap-custom.css" media="screen">-->
```
Comment (slm4996) [inline include/client/header.inc.php]: Uncomment this to see another theme in action.

**Comment by Chefkeks** [comment]
Even though I currently have not the time to try that pull request, I like it in which direction this is going and screenshots don't look bad either - so keep up the good work! ;)

**Comment by greezybacon** [comment]
@slm4996 I'd ike to see this rebased against the `develop-next` branch. We made some major changes to the client portal, and I would like to avoid reworking your theme as much as possible with we collapse `develop-next` into `develop`. Also I don't like the navigation bar appearance. I think @protich would like to see it somehow go away completely. Perhaps "Open a new ticket" is unnecessary as we already have a button for it. "Check ticket status" is duplicated with a button and with the "Sign In" link. And the knowledgeable is redesigned in `develop-next` to put much more presence on the front page.

**Comment by Chefkeks** [comment]
@greezybacon
I agree with the following: "protich would like to see it somehow go away completely".

Would surely improve the look of the client portal in my eyes ;)

**Comment by balojs** [comment]
:+1:

**Comment by slm4996** [comment]
@greezybacon and @protich and everyone else, what do you think of this version running lice on develop-next? http://osticket.ssdev.info/ I haven't pushed it out to my repo yet, pending feedback.

**Comment by greezybacon** [comment]
I like the direction!

My original concept was to deemphasize the "Open a New Ticket" button a bit (from the current version of osTicket) in favor of the knowledge base articles. Maybe you could make the padding smaller, and/or drop the background color except for hover, and add some spacing between that and the sidebar below it, and make the "Featured Questions" a bit more emphasized.

I also think that somehow the top banner will need to have a stark white background, because so many users are somehow unable to find a logo with a transparent background.

**Comment by Chefkeks** [comment]
Looking really good!

Agree with Jared's suggestions to further improve it. Especially the transparent color of custom logos is an possible future issue source and maybe an option in osTicket to allow the admin to make the image transparent would be really awesome, but I don't have any clue how hard it's to realize that, but when it's not that hard, I think it's definitely a must-have feature for the upcoming themes - hope this gives some inspiration @greezybacon ;)

**Comment by slm4996** [comment]
Time for some more feedback please. I think I like the bootstrap panels for the FAQ content in the sidebar, but think it might be too big for the main area FAQs. Thoughts? http://osticket.ssdev.info

**Comment by Chefkeks** [comment]
Just looked at the theme from my smart phone, looks pretty good, but would make the "open new ticket" button in green and the entries under the "staff questions" also a bit different since I did not notice that there were 2 entries the first time I looked at the page.

And the --- menu button on top is good, but when it expands the links are in one row. Would be better to have one row for each link (support center home, knowledge base, guest user)

So much about the mobile theme, but all in all really good work and appreciate it - I'll take a look at the desktop version the next time I'm on the desktop.

Cheers,
Michael

**Comment by slm4996** [comment]
That seems to be one area of disagreement, the open ticket button. I can see both sides. Highlight it and make it super easy to find, that is after all the reason for the software, or make it passive and try to focus on the knowledgebase answers first so fewer tickets get opened. 

Maybe this needs to be an admin interface option?

**Comment by slm4996** [comment]
I've updated the knowledge base pages, and redone the navbar.

**Comment by Chefkeks** [comment]
So, again I take a look at the mobile version (guess I spend more time on the phone at home after work than on work at the desktop :D)

Have to admit the old navbar with the following button:
--------
--------
--------
looked better, only the extra click to get the sub-options ("Sign in" and "Check ticket status") under the "Guest User" are not mobile friendly. Maybe display them directly? At the moment an extra click is necessary to see the 2 sub-options of the "Guest user".

I personally like the more emphasized "new ticket button", but I'm still not happy with the "staff questions"...

I like that the 2 questions itself are now more highlighted, but it would be great to not de-emphasize the category "staff questions" that much. Maybe use the current blue color for the "staff questions" and a lighter blue for the questions.

Sorry for the extra work, but really appreciate your work and know that designing/themeing can be really hard and is sometimes a long process (I've been through myself several times), so keep up the good work :+1:

**Comment by Chefkeks** [comment]
Just looked at the desktop version.

What I would like to improve is that "Guest User - Check Ticket Status - Sign in" menu, since it's too I think.

So following my suggestions - prepared a few once so you can have a look ;)

Suggestions 1:
Remove "Guest User" and just show "Check Ticket Status" and "Sign in" and emphasize the "Staff questions" a little bit more:
image


Suggestion 2:
Remove "Guest User" and remove "Check Ticket Status". Just show "Sign in" and emphasize the "Staff questions" a little bit more:
image

This suggestion is leading in the direction that the pages to "Check Ticket Status" and "Sign in" should may be combined, so they can see the "check ticket status" on the left and "Sign in" on the right (thinking of a table).

Maybe it would also be possible to go a step further and have just 1 login form which allows both options, so that you can either enter "Username or Email" and "Password or Ticket number". Behind then there should be some logic, that if the user has an account already, but is signing in with the "Username" and the "ticket number" (as far as I know that combo is currently not possible?!), it will should then may just display only ticket itself, but asks the user for the password of his account to show all of his/her tickets. Or when the user hits the button/option "Tickets" (or "Show all tickets" or "My tickets") request the password then. 

I personally would really like to have just 1 "Sign In" form and I guess there is some potential here to improve that. Reason is that I just think, that currently these 2 login forms are a bit confusing when you're a user who e.g. access osTicket web portal for the first time / not every day to check your tickets' status.

Finally, an Icon before "Sign in" e.g. http://osticket.ssdev.info/assets/default/images/lock.png would be cool. Unfortunately I wasn't able to modify the code that way with my browsers developers tools (from which I took the screenshots with)

Hope I could give you an idea and would love to see some cosmetic changes at the web portal coming with a theme by @slm4996, but may built-in right into osTicket by @greezybacon ;)

Cheers,
Michael

**Comment by rp-tech** [comment]
I can't WAIT for this feature!

**Comment by slm4996** [comment]
I just pulkled the latest develop-next changes and made a new pull request for that branch: #2165 All work will continue there.

**Pull request closed without merging**